mmdet/models/losses/iou_loss.py

```diff
@@ -69,6 +69,60 @@ def bounded_iou_loss(pred, target, beta=0.2, eps=1e-3):
     return loss
 
 
+@weighted_loss
+def giou_loss(pred, target, eps=1e-6):
+    """GIoU loss.
+
+    https://github.com/sfzhang15/ATSS/blob/master/atss_core/modeling/rpn/atss/loss.py#L36
+
+    Args:
+        pred (Tensor): Predicted bboxes of format (x1, y1, x2, y2),
+            shape (n, 4).
+        target (Tensor): Corresponding gt bboxes, shape (n, 4).
+        eps (float): Eps to avoid log(0).
+
+    Return:
+        Tensor: Loss tensor.
+    """
+
+    pred_x1 = pred[:, 0]
+    pred_y1 = pred[:, 1]
+    pred_x2 = pred[:, 2]
+    pred_y2 = pred[:, 3]
+    pred_x2 = torch.max(pred_x1, pred_x2)
+    pred_y2 = torch.max(pred_y1, pred_y2)
+    pred_area = (pred_x2 - pred_x1) * (pred_y2 - pred_y1)
```

Comment:
This is a convention in mmdetection. For two coordinates x1, x2, the real length should be x2 - x1 + 1.
You can refer [`bbox_overlaps`](https://github.com/open-mmlab/mmdetection/blob/e907139cb8f4c0195abcf31101925819f369ecc1/mmdet/core/bbox/geometry.py#L63) function.